scipy/optimize/tests/test_linesearch.py

```diff
@@ -190,9 +192,11 @@ def test_line_search_wolfe2(self):
             f0 = f(x)
             g0 = fprime(x)
             self.fcount = 0
-            s, fc, gc, fv, ofv, gv = ls.line_search_wolfe2(f, fprime, x, p,
-                                                           g0, f0, old_f,
-                                                           amax=smax)
+            with warnings.catch_warnings():
+                warnings.simplefilter('ignore', RuntimeWarning)
+                s, fc, gc, fv, ofv, gv = ls.line_search_wolfe2(f, fprime, x, p,
+                                                               g0, f0, old_f,
+                                                               amax=smax)
```

Comment:
I guess you could only filter `LineSearchWarning` (`from scipy.optimize.linesearch`) because plain `RuntimeWarning`s are not expected here IIRC.